### PR TITLE
display vulns in package.json

### PR DESCRIPTION
**Pull request by bmvermeer**

### CODE REVIEW

File: src/getImports/packageJsonParser.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  const packages = [];
+
+  const lines = [];
+  source.split(/\r?\n/).forEach(function(line) {
+    lines.push(line);
+  });
+
+  const pjson = JSON.parse(source);
+
+  for (var dep in pjson.dependencies) {
```
Comment (thisislawatts) [inline src/getImports/packageJsonParser.js:11]: Should we also be scanning folks [devDependencies](https://docs.npmjs.com/files/package.json#devdependencies) and [peerDependencies](https://docs.npmjs.com/files/package.json#peerdependencies)?

**Comment by bmvermeer** [comment]
Maybe in the future. I dont believe it is good to do this by default. I think it will be considered noise but lets monitor this. 
Maybe we should make it optional just like in the CLI.

File: src/getImports/packageJsonParser.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    var p = {};
+    p.fileName = fileName;
+    p.name = dep;
+    p.loc = findLoc(dep, lines);
+    p.line = p.loc.start.line;
```
Comment (thisislawatts) [inline src/getImports/packageJsonParser.js:16]: I think we're missing the `version` property. Which looks is being returned by the `htmlParser`:
https://github.com/snyk/vulncost/blob/77dc018fb3282cbac05486389ef0a8855ffc643e/src/getImports/htmlParser.js#L91

We should also aim to validate the version string with `semver.valid`, to filter out instances where users are directly referencing GitHub repos (although we could also test these 🤔).

Here's a gist based on the npm documentation for permitted dep declaration styles, filtering out those which can't be resolved to a version.
https://gist.github.com/thisislawatts/48e5c6bdbdaa4156e18bd81f60380a06

Ideally we should use either the yarn.lock/package-lock.json which gives a more accurate idea of what will be installed during deployment.

Comment (bmvermeer) [reply src/getImports/packageJsonParser.js:16]: Check the js / ts parser. By default we take the latest version if no manufest is available or else we check the version in the manifest.

For html when using a cdn we do not depend on a manifest file but the version inline.

**Comment by thisislawatts** [comment]
@bmvermeer I am not sure I agree, given the user is in the manifest file and seeing Snyk annotations on their `dependencies` the absence of annotations on the `devDependencies` does not clearly communicate these are being ignored by default. Instead my assumption would be that these devDeps should be considered safe/secure.

**Comment by thisislawatts** [comment]
@bmvermeer In case my earlier edit got missed we should definitely filter out any packages which are not being loaded from NPM, this includes the GitHub & local style mentioned here:
https://docs.npmjs.com/files/package.json#github-urls

**Pull request merged**